private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisher.java

```diff
@@ -93,14 +93,15 @@ public static void main(String[] args)
 
     try {
       List<CompletableFuture<Void>> futures = new ArrayList<>();
-      futures.add(upload(repo, credentials, coords, ".pom", pom, gpgSign));
 
       if (mainArtifact != null) {
         String ext =
             com.google.common.io.Files.getFileExtension(mainArtifact.getFileName().toString());
         futures.add(upload(repo, credentials, coords, "." + ext, mainArtifact, gpgSign));
       }
 
+      futures.add(upload(repo, credentials, coords, ".pom", pom, gpgSign));
```

Comment:
Relevant discussion for this change:
https://github.com/vaticle/bazel-distribution/pull/318

Might break this out into a separate contribution since this is unrelated, yet necessary for publishing support for Artifactory repositories.